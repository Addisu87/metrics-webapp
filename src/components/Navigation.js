import React from 'react';
import { NavLink } from 'react-router-dom';

function Navigation() {
  return (
    <nav className="navbar">
      <div className="navTitle">
        <h1>Covid-19 Tracker</h1>
      </div>
      <ul className="navLinks">
        <li className="navLink">
          <NavLink to="/">Home</NavLink>
        </li>
        <li className="navLink">
          <NavLink to="/continents">Continents</NavLink>
        </li>
      </ul>
    </nav>
  );
}

export default Navigation;
