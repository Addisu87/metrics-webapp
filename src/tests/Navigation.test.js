import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import Navigation from '../components/Navigation';
import store from '../redux/configureStore';

describe('Jest Snapshot testing suite', () => {
  it('Matches DOM Snapshot', () => {
    const { container } = render(
      <MemoryRouter>
        <Provider store={store}>
          <Navigation />
        </Provider>
      </MemoryRouter>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
