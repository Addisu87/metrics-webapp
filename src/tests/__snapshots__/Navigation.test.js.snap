// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Jest Snapshot testing suite Matches DOM Snapshot 1`] = `
<nav
  class="sc-bczRLJ dwLGhv"
>
  <div
    class="sc-gsnTZi jyTtBf"
  >
    <a
      aria-current="page"
      class="sc-dkzDqf gedDWv active"
      href="/"
    >
      <svg
        class="sc-hKMtZM la-difT"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 504C119 504 8 393 8 256S119 8 256 8s248 111 248 248-111 248-248 248zM142.1 273l135.5 135.5c9.4 9.4 24.6 9.4 33.9 0l17-17c9.4-9.4 9.4-24.6 0-33.9L226.9 256l101.6-101.6c9.4-9.4 9.4-24.6 0-33.9l-17-17c-9.4-9.4-24.6-9.4-33.9 0L142.1 239c-9.4 9.4-9.4 24.6 0 34z"
        />
      </svg>
    </a>
    <h2>
      2022
    </h2>
  </div>
</nav>
`;
